.circleci/config.yml

```diff
@@ -1,72 +1,125 @@
-    - name: Install and run black for notebooks
-      run: |
-        python -m pip install --upgrade pip
-        python -m pip install black[jupyter]
-        black --check --diff --verbose Python
-  test:
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
-        python-version: ['3.7', '3.10']
-    machine: 
+    machine:
       image: ubuntu-2004:current
-    resource_class: large
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions/cache@v3
-      id: cache
-      with:
-        path: |
-          Data
-        key: notebook-data-${{ hashFiles('Data/manifest.json') }}
-        restore-keys: |
-          notebook-data-${{ hashFiles('Data/manifest.json') }}
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v4
-      with:
-        python-version: ${{ matrix.python-version }}
-    # on windows, the pyenchant package includes enchant
-    - name: Install enchant on non windows systems
-      shell: bash
-      run: |
-        if [ "$RUNNER_OS" == "Linux" ]; then
-          sudo apt-get update
-          sudo apt-get install enchant-2
-        elif [ "$RUNNER_OS" == "macOS" ]; then
-          brew update
-          brew install enchant
-        fi
-    - name: Install dependencies
-      run: |
-        pip install --upgrade pip
-        pip install -r requirements_dev.txt
-        jupyter nbextension enable --py --sys-prefix widgetsnbextension
-    - name: Install SimpleITK from test PyPi from manually launched workflow
-      if: ${{ inputs.testPyPi }}
-      run: pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple SimpleITK
-    - name: Download data
-      if: steps.cache.outputs.cache-hit != 'true'
-      run: python Utilities/downloaddata.py Data/ Data/manifest.json
-    - name: run the test
-      env: 
-        SIMPLE_ITK_MEMORY_CONSTRAINED_ENVIRONMENT: 1
-      run: pytest -v --tb=short tests/test_notebooks.py::Test_notebooks::test_python_notebook
+      - checkout
+      - run:
+          name: Set up Python 3.9
+          command: |
+            sudo apt-get update
+            sudo apt-get install python3.9
+            sudo apt install python3.9-venv
+            python3.9 -m venv env
+            source env/bin/activate
+      - run:
+          name: Install and run black for notebooks
+          command: |
+            source env/bin/activate
+            pip install --upgrade pip
+            pip install black[jupyter]
+            black --check --diff --verbose Python
 
+  test:
+    parameters:
+      os:
+        type: executor
+      python-version: 
+        type: string
+    executor: << parameters.os >>
+    steps:
+      - checkout
+      - restore_cache:
+          key: notebook-data-${{ checksum 'Data/manifest.json' }}
+      - run:
+          name: Install Python << parameters.python-version >>
+          command: |
+            if [ "$RUNNER_OS" == "macos" ]; then
```

Comment:
As the macos and linux share most of the code, possibly simplify as follows:
```
            if [ "$RUNNER_OS" == "macos" ]; then
              brew install pyenv
              eval "$(pyenv init -)"
            fi
            if [ "$RUNNER_OS" == "macos" || "$RUNNER_OS" == "linux" ]; then
              pyenv install << parameters.python-version >> || echo "Python version is already installed."
              pyenv global << parameters.python-version >>
              export PATH="$HOME/.pyenv/versions/<< parameters.python-version >>/bin:$PATH"
              echo -e '\neval "$(pyenv init -)"' >> ~/.bash_profile
              source ~/.bash_profile
            elif [ "$RUNNER_OS" == "windows" ]; then
              choco install python --version=<< parameters.python-version >> --allow-downgrade -y
              python -m pip install --upgrade pip
            fi
```

Reply:
Oh yeah. Thanks for that refactor.